pkg/controller/storagecluster/storagecluster.go

```diff
@@ -774,6 +774,12 @@ func (c *Controller) manage(
 		}
 	}
 
+	if cluster.Status.Phase == "" && cluster.Spec.CloudStorage.MaxStorageNodesPerZone == nil {
```

Comment:
We should not set it if `cloudStorage.MaxStorageNodes` is set. We know that the user is restricting the cluster size to something, so that should do.